slyd/slyd/orm/base.py

```diff
@@ -0,0 +1,550 @@
+            raise ValueError("'path' option must be a string or None.")
+        self.owner = getattr(meta, 'owner', False)
+        if self.owner != False and not isinstance(self.owner, string_types):
+            raise ValueError("'owner' option must be a string or False.")
+        if self.owner and not isinstance(model._fields.get(self.owner),
+                                         BaseRelationship):
+            raise ValueError("'owner' option must be a relationship name.")
+        self.envelope = getattr(meta, 'envelope', False)
+        if not isinstance(self.envelope, bool):
+            raise ValueError("'envelope' option must be a boolean.")
+        self.envelope_remove_key = getattr(meta, 'envelope_remove_key', False)
+        if not isinstance(self.envelope_remove_key, bool):
+            raise ValueError("'envelope_remove_key' option must be a boolean.")
+
+
+class ModelMeta(type):
+    """Meta class for models"""
+    def __new__(mcs, name, bases, attrs):
+        parents = [b for b in bases if isinstance(b, ModelMeta)]
+        if not parents:
+            return super(ModelMeta, mcs).__new__(mcs, name, bases, attrs)
+
+        # check if a model with the same name exists in the registry
+        if name in models:
+            raise ImproperlyConfigured(
+                u"A Model named '{}' already exists".format(name))
+
+        meta = attrs.pop('Meta', None)
+        primary_key = None
+        fields = {}
+        basic_attrs = {}
+        file_schema_attrs = {}
+
+        for attrname, value in iteritems(attrs):
+            if isinstance(value, BaseRelationship):
+                fields[attrname] = value
+            elif isinstance(value, Field):
+                if value.primary_key:
+                    if primary_key:
+                        raise ImproperlyConfigured(
+                            u"Model '{}' declared with more than one primary "
+                            u"key".format(name))
+                    primary_key = attrname
+                fields[attrname] = value
+            # move decorated marshmallow methods to the file schema
+            elif hasattr(value, '__marshmallow_tags__'):
+                file_schema_attrs[attrname] = value
+            else:
+                basic_attrs[attrname] = value
+
+        if fields and not primary_key:
+            raise ImproperlyConfigured(
+                u"Model '{}' declared with no primary key".format(name))
+
+        dependencies = {}
+        for attrname, field in iteritems(fields):
+            if attrname == primary_key:
+                dependencies[attrname] = set()
+            elif isinstance(field, DependantField):
+                dependencies[attrname] = {primary_key, field.when}
+            else:
+                dependencies[attrname] = {primary_key}
+        try:
+            ordered_fields = toposort_flatten(dependencies)
+        except ValueError as e:
+            raise ImproperlyConfigured(e.message)
```

Comment:
This portion of code checks dependencies are acyclic and created the `ordered_fields` to attach it the class right? If this is the case, can you refactor this into a helper function so that we have only `cls._ordered_fields = self._build_ordered_fields`?